master/internal/populate_metrics.go

```diff
@@ -0,0 +1,321 @@
+	if err != nil {
+		return err
+	}
+
+	validationBatchMetrics := []*structpb.Struct{}
+
+	for j := 0; j < stepSize; j++ {
+		validationBatchMetrics = append(validationBatchMetrics, makeMetrics())
+	}
+
+	validationMetrics := trialv1.TrialMetrics{
+		TrialId:        trialID,
+		StepsCompleted: stepSize,
+		Metrics: &commonv1.Metrics{
+			AvgMetrics:   makeMetrics(),
+			BatchMetrics: validationBatchMetrics,
+		},
+	}
+
+	_, err = api.ReportTrialValidationMetrics(ctx,
+		&apiv1.ReportTrialValidationMetricsRequest{
+			ValidationMetrics: &validationMetrics,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// PopulateExpTrialsMetrics adds metrics for a trial and exp to db.
+func PopulateExpTrialsMetrics(pgdb *db.PgDB, masterConfig *config.Config) error {
+	system := actor.NewSystem("mock")
+	ref, _ := system.ActorOf(sproto.AgentRMAddr, actor.ActorFunc(
+		func(context *actor.Context) error {
+			switch context.Message().(type) {
+			case sproto.DeleteJob:
+				context.Respond(sproto.EmptyDeleteJobResponse())
+			}
+			return nil
+		}))
+	mockRM := actorrm.Wrap(ref)
+	api := &apiServer{
+		m: &Master{
+			trialLogBackend: pgdb,
+			system:          system,
+			db:              pgdb,
+			taskLogBackend:  pgdb,
+			rm:              mockRM,
+			config:          masterConfig,
+			taskSpec:        &tasks.TaskSpec{},
+		},
+	}
+
+	_, err := user.UserByUsername("admin")
+	if err != nil {
+		return err
+	}
+
+	resp, err := api.Login(context.TODO(), &apiv1.LoginRequest{Username: "admin"})
+	if err != nil {
+		return err
+	}
+
+	ctx := metadata.NewIncomingContext(context.TODO(),
+		metadata.Pairs("x-user-token", fmt.Sprintf("Bearer %s", resp.Token)))
+
+	// create exp and config
+	maxLength := expconf.NewLengthInBatches(100)
+	maxRestarts := 0
+	activeConfig := expconf.ExperimentConfig{ //nolint:exhaustivestruct
+		RawSearcher: &expconf.SearcherConfig{ //nolint:exhaustivestruct
+			RawMetric: ptrs.Ptr("loss"),
+			RawSingleConfig: &expconf.SingleConfig{ //nolint:exhaustivestruct
+				RawMaxLength: &maxLength,
+			},
+		},
+		RawEntrypoint:      &expconf.Entrypoint{RawEntrypoint: "model_def:SomeTrialClass"},
+		RawHyperparameters: expconf.Hyperparameters{},
+		RawCheckpointStorage: &expconf.CheckpointStorageConfig{ //nolint:exhaustivestruct
+			RawSharedFSConfig: &expconf.SharedFSConfig{ //nolint:exhaustivestruct
+				RawHostPath: ptrs.Ptr("/"),
+			},
+		},
+		RawMaxRestarts: &maxRestarts,
+	} //nolint:exhaustivestruct
+	activeConfig = schemas.WithDefaults(activeConfig)
+	model.DefaultTaskContainerDefaults().MergeIntoExpConfig(&activeConfig)
+
+	var defaultDeterminedUID model.UserID = 2
+	exp := &model.Experiment{
+		JobID:                model.NewJobID(),
+		State:                model.ActiveState,
+		Config:               activeConfig.AsLegacy(),
+		StartTime:            time.Now(),
+		OwnerID:              &defaultDeterminedUID,
+		ModelDefinitionBytes: []byte{},
+		ProjectID:            1,
+	}
+	err = pgdb.AddExperiment(exp, activeConfig)
```

Comment:
it'd be great to attach a screenshot to the PR with the metrics charts as a reference.

Reply:
<img width="1286" alt="Screen Shot 2023-03-29 at 9 43 23 PM" src="https://user-images.githubusercontent.com/26676264/228707115-d9c1bb64-5b59-4685-a716-54d72cebe2b2.png">

<img width="1299" alt="Screen Shot 2023-03-29 at 9 43 46 PM" src="https://user-images.githubusercontent.com/26676264/228707109-d9f9a4cd-daac-434d-a944-a28f47f74b49.png">

<img width="1301" alt="Screen Shot 2023-03-29 at 9 47 35 PM" src="https://user-images.githubusercontent.com/26676264/228707099-9f167e2c-3224-429f-920e-d010c29ab68e.png">